### PR TITLE
Week 11, Happy-Thoughts, Maria Thomasson

### PR DESCRIPTION
**Pull request by MariaThomasson**
Link to my deployed project: https://write-happy-thoughts.netlify.app/

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/utils/Emoji.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    <span
+        className="emoji"
+        role="img"
+        aria-label={props.label ? props.label : ""}
+        aria-hidden={props.label ? "false" : "true"}
+    >
+        {props.symbol}
+    </span>
+);
+export default Emoji;
```
Comment (nehrwein) [inline code/src/utils/Emoji.js:13]: I like the approach to make a emoji-component!

File: code/src/components/Loading.css [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  }
+  45% {
+    transform: scale(1);
+  }
+  60% {
+    transform: scale(0.95);
+  }
+  100% {
+    transform: scale(0.9);
+  }
```
Comment (nehrwein) [inline code/src/components/Loading.css:85]: The loading-screen isn't working for me

File: code/src/components/SubmitButton.css [hunk 3]
```diff
@@ -0,0 +1,28 @@
+@import url('https://fonts.googleapis.com/css2?family=Roboto+Mono&display=swap');
+
+
+* {
+  font-family: 'Roboto Mono', monospace;
+  text-rendering: optimizeLegibility;
+}
```
Comment (nehrwein) [inline code/src/components/SubmitButton.css:7]: What does` text-rendering: optimizeLegibility` do? :)

File: code/src/components/SubmitButton.css [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    
+}
+
+.submit-button:hover {
+  transform: scale(1.1);
+}
+
+.emoji {
+padding: 4px;
+}
```
Comment (nehrwein) [inline code/src/components/SubmitButton.css:28]: This makes your like-buttons oval. If you take away the padding on the emoji, you will get a circle

File: code/src/index.css [hunk 5]
```diff
@@ -1,13 +1,88 @@
...
+}
+
+.form-text {
+  box-sizing: border-box;
+  width: 100%;
+  margin: 10px 0;
+  padding: 15px;
+  resize: none;
+}
+
```
Comment (nehrwein) [inline code/src/index.css:50]: .form-text doesn't seem to be applies in your html. The styles should be applied to form textarea, I guess :)

**Review by nehrwein — COMMENTED** [review]
hej Maria,
nice job! You did a great happy-thoughts-app and it looks good. I left some comments  in the code, maybe it helps with minor issues.
//Birgit

File: README.md [hunk 6]
```diff
@@ -1,13 +1,14 @@
...
-Replace this readme with your own information about your project.
-
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+In this week's project, we practice our React state skills by fetching and posting data to an API. 
+Technigo built a simple API to collect 'happy thoughts'. Think of it as our own version of Twitter, but with less negativity, and 100% fewer US presidents using it.
 
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+Using Emoji in React article: https://medium.com/@seanmcp/%EF%B8%8F-how-to-use-emojis-in-react-d23bbf608bf7
```
Comment (Rephili) [inline README.md:8]: Great that you also add the links :)

File: code/src/components/Header.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+
+const Header = () => 
+<h1>Happy thoughts 
+<span 
+role="img"
+aria-label="love letter emoji"
+className="h1-span"> {'💌 '}
```
Comment (Rephili) [inline code/src/components/Header.js:8]: Hadn't seen it before like this :)

File: code/src/components/Loading.css (hunk 2 above)
Comment (Rephili) [reply code/src/components/Loading.css:85]: Can't see it either

File: code/src/components/NewThought.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+return (
+  <form onSubmit={onFormSubmit}>
+  <label htmlFor="newThought">What is making you happy right now?</label>
+  <textarea
+  id="newThought"
+  type="text" 
+  name="Text"
+  rows="4"
+  cols="50"
+  spellCheck="true"
```
[the remark this answers is not in the document]
Comment (Rephili) [reply code/src/components/NewThought.js:14]: Great advice Birgit :)

File: code/src/utils/Emoji.js (hunk 1 above)
Comment (Rephili) [reply code/src/utils/Emoji.js:13]: I agree with Birgit :)

File: code/src/components/NewThought.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  required
+  placeholder="Write a happy thought here..."
+  value={newThought} 
+  onChange={(event) => setNewThought(event.target.value)} 
+  />
+  <p className='character-counter'>{newThought.length}/140</p>
```
Comment (Rephili) [inline code/src/components/NewThought.js:20]: Nice!

**Review by Rephili — COMMENTED** [review]
Hej Maria! Well done! Since Birgit already commented on your code I didn't have much to add to it. You may consider having a look at the loading-screen (not a requirement though) to make sure it works. :)

**Pull request closed without merging**